pkg/logs/docker/docker.go

```diff
@@ -41,5 +44,56 @@ func ParseMessage(msg []byte) (string, string, []byte, error) {
 	}
 	to += messageHeaderLength
 	ts := string(msg[messageHeaderLength:to])
+
 	return ts, status, msg[to+1:], nil
+
+}
+
+// getPreMessageLength finds length of the 8 byte header, timestamp, and space
+// the is in front of each 16Kb chunk of message
+func getPreMessageLength(msg []byte) int {
+	idx := bytes.Index(msg, []byte{' '})
+	if idx == -1 {
+		return 0
+	}
+	return idx + 1
+}
+
+func min(a, b int) int {
+	if a < b {
+		return a
+	}
+	return b
+}
+
+// removePartialHeaders removes the 8 byte header, timestamp, and space
+// that occurs between each 16Kb section of a log greater than 16 Kb in length.
+// If a docker log is greater than 16Kb, each 16Kb section "PartialMessage" will
+// have a header, timestamp, and space in front of it.  For illustration,
+// let's call this "HeadTs ".  For example, a message that is 35kb will be of the form:
+// `HeadTs PartialMessageHeadTs PartialMessageHeadTs PartialMessage`
+// This function removes the "HeadTs " between two PartialMessage sections while
+// leaving the very first "HeadTs "
+func removePartialHeaders(msg []byte) []byte {
+	preMessageLength := getPreMessageLength(msg)
```

Comment:
One quick optimization, we don't need to compute the `PreMessageLength` if the whole message is smaller than `maxDockerBufferSize`, so what about doing this first check first?